### ISSUE TITLE
Error out / better description when there is no components listed.

### ISSUE THREAD

**Issue by cdrage** [redhat-developer/odo#319]
Ran into this earlier today when I didn't have an application set:

```sh

github.com/redhat-developer/ocdev  create-components-same-name ✗                                                                                                                                                                                                                                                                                                   3h40m ⚑  
▶ ./ocdev application list
ACTIVE   NAME
         app
         foobar

github.com/redhat-developer/ocdev  create-components-same-name ✗                                                                                                                                                                                                                                                                                                  3h40m ⚑  ⍉
▶ make bin && ./ocdev update -v --git https://github.com/openshift/nodejs-ex.git

go build -ldflags="-w -X github.com/redhat-developer/ocdev/cmd.GITCOMMIT=b99df98" -o ocdev main.go
DEBU[0000] getOcBinary - searching for oc binary
DEBU[0000] envKubectlPluginCaller =

DEBU[0000] envOcBin =

DEBU[0000] using oc from PATH
DEBU[0000] using oc from %s/usr/local/bin/oc
DEBU[0000] Trying to connect to server 192.168.42.113:8443 - %!v(MISSING)
DEBU[0000] Server https://192.168.42.113:8443 is up
DEBU[0000] isLoggedIn err:  <nil>
 output: "system:admin\n"
DEBU[0000] myproject-
DEBU[0000] Getting BuildConfig:
unable to update the component: unable to get the BuildConfig file: unable to get BuildConfig : resource name may not be empty

```

We should be checking to see if the component is actually available (not blank / is available) and error out better / notify the user that no component is actually selected / there are none / there is no actual application selected either.

**Comment by mik-dass** [comment]
@cdrage @kadel @containscafeine Should the error message display one error message or multiple depending upon the type of the error like no application exists or no application is set as active?

**Comment by kadel** [comment]
In the situation shown above, the message should warn the user about the component. 
It should show display that there is no active component.

**Comment by mik-dass** [comment]
@kadel What if no application or component is present and the user executes the command?

**Comment by kadel** [comment]
no component also means no active component, so that is the same.

If there is no active application then it should display a message informing the user about that.
But currently that is harder to check, you will probably need changes in https://github.com/redhat-developer/odo/pull/358/files for that.

**Comment by mik-dass** [comment]
@kadel and what about a empty project with no applications in it?

**Comment by mik-dass** [comment]
@kadel Ok got it but should the error message in both cases be same or different?

**Comment by kadel** [comment]
different

**Comment by mik-dass** [comment]
@kadel Can we close this?

**Comment by kadel** [comment]
fixed in #374
